turbo/stages/headerdownload/header_algos.go

```diff
@@ -121,6 +121,12 @@ func (hd *HeaderDownload) ReportBadHeader(headerHash libcommon.Hash) {
 	hd.lock.Lock()
 	defer hd.lock.Unlock()
 	hd.badHeaders[headerHash] = struct{}{}
+	hd.UnlinkHeader(headerHash)
```

Comment:
I think we should not call it here as we are anyway calling it in `HeadersUnwind` func of `Unwind`stage_headers.go` to cover for non-bad block cases